shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/metadata/dialect/H2DataSourceMetaData.java

```diff
@@ -29,27 +29,27 @@
  */
 @Getter
 public final class H2DataSourceMetaData implements MemorizedDataSourceMetaData {
-    
+
     private static final int DEFAULT_PORT = -1;
-    
+
     private final String hostName;
-    
+
     private final int port;
-    
+
     private final String catalog;
-    
+
     private final String schema;
-    
+
     private final Pattern pattern = Pattern.compile("jdbc:h2:(mem|~)[:/]([\\w\\-]+);?\\S*", Pattern.CASE_INSENSITIVE);
-    
-    public H2DataSourceMetaData(final String url) {
+
+    public H2DataSourceMetaData(final String url, final String username) {
         Matcher matcher = pattern.matcher(url);
         if (!matcher.find()) {
             throw new UnrecognizedDatabaseURLException(url, pattern.pattern());
         }
         hostName = "";
         port = DEFAULT_PORT;
         catalog = matcher.group(2);
-        schema = null;
+        schema = username;
```

Comment:
Does the change correct?
If the proposal of this issue is adding more test cases, why change the production code?